src/Components/getList.js

```diff
@@ -11,11 +24,25 @@ class GetList extends React.Component {
           return isLoading ? (
             <div>loading...</div>
           ) : (
-            <div>
+            <div className="grocery-list">
               <h1>Groceries</h1>
-              <ul>
+              <ul className="tl f2">
                 {data.map(item => (
-                  <li key={item.id}>{item.itemName}</li>
+                  <li className="bg-white pa1 shadow" key={item.id}>
+                    <label>
+                    <input 
+                      type="checkbox"
+                      ischecked={this.state.toggleCheck}
+                      defaultChecked={
+                        item.lastPurchasedDate
+                          ? Date.now() / 1000 - item.lastPurchasedDate.seconds <
+                            86400
+                          : false
+                      }
```

Comment:
I like your approach, checking this in the component level I think it's a good call!

One suggestion though, here, we have 2 [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) - `1000` and `86400`. I strongly suggest moving those into constants outside of the component (line 4) and give them a meaningful name, for example:

`86400` could become something like `const SECONDS_IN_ONE_DAY = 86400;`, as for `1000` I really don't understand what it is 😅.

If you want to take things even further, I'd suggest creating a helper function that help you calculate this and that's easier to reason about than this whole operation, something like this:

```js
const wasItemPurchasedLast24Hours = (purchasedDate) => purchasedDate ? ... : ....;
```

If you have that, you can just do `defaultChecked={wasItemPurchasedLast24Hours(theItem)`, that's way easier to reason about than a _random_ date operation.

Reply:
I wouldn't worry about having a long name for variables/function names, that's why we have IDEs/Text Editors that can take care of autocompletion.

A long descriptive name, is better than a short difficult to understand name!

cc y'all @Tanzi11 @meganmckissack @nabilkaz @DeeThompson Since I think this is a good lesson to learn!

Reply:
Thank you @alejandronanez ! I will refactor this, that's a good idea!